--- a/src/main/java/org/springframework/samples/petclinic/model/Pet.java
+++ b/src/main/java/org/springframework/samples/petclinic/model/Pet.java
@@ -20,6 +20,7 @@
 import org.springframework.beans.support.MutableSortDefinition;
 import org.springframework.beans.support.PropertyComparator;
 import org.springframework.format.annotation.DateTimeFormat;
+import java.math.BigDecimal;
 
 import jakarta.persistence.*;
 import java.time.LocalDate;
@@ -37,6 +38,10 @@
     @Column(name = "birth_date")
     private LocalDate birthDate;
 
+    @Column(name = "weight")
+    @Digits(integer = 8, fraction = 2)
+    private BigDecimal weight;
+
     @ManyToOne
     @JoinColumn(name = "type_id")
     private PetType type;
@@ -77,6 +82,14 @@
         this.birthDate = birthDate;
     }
 
+    public BigDecimal getWeight() {
+        return weight;
+    }
+
+    public void setWeight(BigDecimal weight) {
+        this.weight = weight;
+    }
+
     public PetType getType() {
         return type;
     }
